### PR TITLE
Flutter Web support

### PR DESCRIPTION
**Pull request by jamesdaniels**

### CODE REVIEW

File: src/frameworks/flutter/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+function checkForFlutterCLI() {
+  const flutter = execSync(`flutter --version`, { stdio: "ignore" });
+  if (flutter.status) throw new FirebaseError("Flutter CLI not found.");
+}
+
+export function build(cwd: string): Promise<BuildResult> {
+  checkForFlutterCLI();
+  const build = execSync(`flutter build web --release`, {
```
Comment (kevmoo) [inline src/frameworks/flutter/index.ts:31]:
```suggestion
  const build = execSync(`flutter build web`, {
```

`--release` is the default

Comment (kevmoo) [reply src/frameworks/flutter/index.ts:31]: How is customization handled here? Folks might want to pass a number of things to the CLI...

Comment (jamesdaniels) [reply src/frameworks/flutter/index.ts]: Fair q. 

Short term, we have a target environment variable landing soon e.g, `FIREBASE_HOSTING_BUILD_TARGET=profile`
Long term, one will be able to override the build command.

Mid term... what are the most common? 

I could see respecting a `FIREBASE_HOSTING_BUILD_ARGS="..."` env variable

**Comment by jamesdaniels** [comment]
Test coverage and changelog coming right up

**Review by leoortizz — APPROVED** [review]
LGTM! I was able to deploy and develop locally with this PR. 

Just left a few nits for small improvements.

File: src/test/frameworks/flutter/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      sandbox
+        .stub(fsPromises, "readFile")
+        .withArgs(join(cwd, "pubspec.yaml"))
+        .returns(
+          Promise.resolve(
+            Buffer.from(`dependencies:
+  flutter:
+    sdk: flutter`)
+          )
+        );
```
[the remark this answers is not in the document]
Comment (jamesdaniels) [reply src/test/frameworks/flutter/index.spec.ts:38]: TIL thanks

**Pull request merged**